docs/architecture/deployment-execution/import.md

```diff
@@ -0,0 +1,184 @@
+(importing-resources)=
+# Importing resources
+
+There are a number of scenarios in which it is necessary to import existing
+resources for management by Pulumi:
+
+* Migrating manually managed resources to an infrastructure-as-code solution;
+* Migrating from other infrastructure-as-code platforms to Pulumi;
+* Moving resources between Pulumi stacks in cases where e.g. `pulumi state move`
+  is not sufficient.
+
+Pulumi offers two approaches for importing resources: the `import` resource
+option and the `pulumi import` command. At a minimum, importing a resource
+involves adding the resource's state to the destination stack's
+[state](state-snapshots). Once the resource has been added to the stack, the
+Pulumi CLI is able to manage the resource like any other.
+
+## The `import` resource option
+
+The [`import` resource
+option](https://www.pulumi.com/docs/iac/concepts/options/import/) accepts the
+[ID](resource-ids) of an existing resource whose state should be
+[](pulumirpc.ResourceProvider.Read) and imported into the stack.
+
+Importing a resource using the `import` resource option requires that the
+desired state described by the Pulumi program for a resource being imported
+matches the actual state of the resource as returned by the provider. More
+precisely, given a resource `R` of type `T` with import ID `X` and set of inputs
+(as specified in the program) `Iₚ`, the engine performs the following sequence
+of operations:
+
+1. Fetch the current inputs `Iₐ` and state `Sₐ` for the resource of type `T`
+   with ID `X` from its provider by calling the provider's
+   [](pulumirpc.ResourceProvider.Read) method. If the provider does not return a
+   value for `Iₐ`, the provider does not support importing resources and the
+   import fails.
+2. Process the [`ignoreChanges` resource
+   option](https://www.pulumi.com/docs/iac/concepts/options/ignorechanges/) by
+   copying the value for any ignored input property from `Iₐ` to `Iₚ`.
+3. Validate the resource's inputs and apply any programmatic defaults by passing
+   `Iₚ` and `Iₐ` to the provider's [](pulumirpc.ResourceProvider.Check) method.
```

Comment:
This renders as empty in GitHub, but it's the right syntax for html rendering?

Can `[Check](pulumirpc.ResourceProvider.Check)` also work?

Reply:
You can supply values between the `[]` to provide alternate text, yes. I've opted not to since if you don't Sphinx defaults to pulling the title, which in the case of RPC calls/services/messages comes with some (IMO?) cute iconography which can help you pick them out at a glance. E.g.:

![image](https://github.com/user-attachments/assets/7e54ac3e-d29d-479f-baf0-298997998483)